lib/_http_server.js

```diff
@@ -187,6 +187,10 @@ ServerResponse.prototype.writeHead = function(statusCode, reason, obj) {
     headers = obj;
   }
 
+  statusCode |= 0;
+  if (statusCode < 100 || statusCode > 999)
+    throw new Error(`Invalid status code: ${statusCode}`);
```

Comment:
@ChALkeR It's debatable, but since `statusCode` is ultimately coerced to an integer value (e.g. 0 for non-numeric values) I thought maybe `RangeError` would fit better since `0 < 100`. \*shrug\*